File: bandform-backend/src/main/java/xyz/wmmp/bandform_backend/data/UserStatus.java
package xyz.wmmp.bandform_backend.data;

public enum UserStatus {
    NOBANDRAND, //Not in band and willing to be selected in random matching
    NOBANDSEL, //Not in band only willing to choose or be chosen.
    BAND, //In band, don't want to be matched again
    BANDRAND, //In band, willing to join another random band
    BANDSEL, //In band, willing to jin anohter band that is seleced.
}

File: bandform-web/frontend/constants/userStatus.ts
import { UserStatus } from "../gql/graphql";

export type UserStatusOption = {
  value: UserStatus;
  label: string;
  description: string;
};

export const NOT_IN_BAND_STATUS_OPTIONS: UserStatusOption[] = [
  {
    value: "NOBANDRAND",
    label: "Auto-match me",
    description:
      "You can be automatically placed into a new band by the band randomizer — no approval needed each time.",
  },
  {
    value: "NOBANDSEL",
    label: "Let me choose",
    description:
      "You'll only join a band by sending or accepting a request or invite yourself.",
  },
];

export const IN_BAND_STATUS_OPTIONS: UserStatusOption[] = [
  {
    value: "BAND",
    label: "Not looking",
    description:
      "You're set on your current band and don't want to be matched elsewhere.",
  },
  {
    value: "BANDRAND",
    label: "Auto-match me",
    description:
      "You can be automatically placed into another band by the randomizer.",
  },
  {
    value: "BANDSEL",
    label: "Let me choose",
    description:
      "You're open to joining another band, but only via a request or invite you approve.",
  },
];
